Add Ruby/Basics exercise (#1043)
exercism/v3: languages/ruby/exercises/concept/basics/.docs/after.md

> Ruby is a dynamic and strong typed language. On dynamic languages the type of a variable/object is resolved at runtime, which means that its value/type can be changed until the very last minute (when it gets parsed by the interpreter).

Reviewer: ```suggestion
Ruby is a dynamic and strongly typed language. In dynamic languages the type of a variable or object is resolved at runtime, which means that its value or type can be changed up to the very last minute (when it gets parsed by the interpreter).
```
My suggestion here is based solely on grammatical concerns. I'm not sure about the latter part of this sentence even with these changes. Can someone with more Ruby nous comment?

Reviewer 2: "the very last moment" rather than "minute".  Ruby may be considered slow, but not quite that slow.

"For dynamic languages..."

Instead of "at runtime" it might be "during run-time" because the type referenced by a variable may change many times during run-time.

Reviewer 3: > Ruby may be considered slow, but not quite that slow. 🤣 Good point

> And what do we mean with strong typed? Once we know the type of variable/object, Ruby is strict about what you can do with it, for example:

Reviewer: ```suggestion
And what do we mean with strongly typed? Once we know the type of a variable or object, Ruby is strict about what you can do with it, for example:
```

```ruby
x = '2'
y = x + 'n'
=>  '2n'
```

**But**

```ruby
x = '2'
y = x + 2
=> TypeError (no implicit conversion of Integer into String)

> And remember, in Ruby everything is an object even classes are instances of the class `Class`, for example:

Reviewer: ```suggestion
Remember, in Ruby everything is an object. Even classes are instances of the class `Class`. For example:
```

> ```Ruby

Reviewer 4: ```suggestion
```ruby
```

1.class
=> Integer

> Integer.instance_of?(Class)
> => true

Reviewer 2: ```suggestion
Integer.class
=> Class
```

This is a more direct example of Integer class being an instance of `Class`.

But if we cant to communicate that everything is an object, then we can do:

```ruby
Integer.is_a?(Object)
=> true
```

Author: I like this:
```
Integer.is_a?(Object)
# => true

Class.is_a?(Object)
# => true
```

Author: I've added a commit doing what I suggested 👍

```

This means that we can also define classes like this:

```ruby
Car = Class.new do
  def run
    'running'
  end
end

Car.new.run
=> 'running'
```

> Finally, bear in mind that the `Integer` object holds values that may be defined as one or more (consecutive) digits and its methods support most of the [default mathematical operators][integers-docs].

Reviewer 2: ```suggestion
Finally, bear in mind that the `Integer` object holds values that may be defined as one or more (consecutive) digits and its methods support many of the [mathematical operators][integers-docs].
```
